lib/graphql/batch/loader.rb

```diff
@@ -13,9 +15,12 @@ def self.load_many(keys)
       self.for.load_many(keys)
     end
 
+    attr_accessor :loader_key
+
     def load(key)
-      if queue.frozen?
-        raise "Loader can't be used after batch load"
+      loader = Executor.current.loaders[loader_key] ||= self
+      if loader != self
+        raise "load called on loader that wasn't registered with executor"
```

Comment:
Just to make sure I understand this, this would occur if you call `a = Loader.for(:something)` and then use `a` from either another thread? Would this also raise across different queries on a single Thread?